lib/sdk/tabs/observer.js

```diff
@@ -39,14 +39,28 @@ const Observer = Class({
     // that before emitting "activate" on selected tab.
     this.on("select", tab => {
       const selected = this[selectedTab];
+
+      // Bug 942511: this[selectedTab] is never updated when a popup window is
+      // closed. This leads to the situation where the properties of ActiveTab 
+      // are undefined because ActiveTab relates to the closed popup window. To 
+      // work around this situation we emit "activate" on the selected tab when
+      // this[selectedTab] was not updated (not equal to null) and when the 
+      // properties of ActiveTab are undefined.
+      const activeTabId = require("../tabs").activeTab.id;
```

Comment:
use this instead please:

    // at the top
    { getActiveTab, getTabs, getTabId } = require("./utils");

    // then use
    const activeTabId = getTabId(getActiveTab());